contracts/Strategy.sol

```diff
@@ -92,46 +115,98 @@ contract Strategy is BaseStrategy {
             _rewards,
             _keeper,
             _bentoBox,
-            _kashiPair
+            _kashiPairs,
+            _pids
         );
 
         emit Cloned(newStrategy);
     }
 
-    function _initializeStrat(address _bentoBox, address _kashiPair) internal {
+    function _initializeStrat(
+        address _bentoBox,
+        address[] memory _kashiPairs,
+        uint256[] memory _pids
+    ) internal {
         require(
-            address(kashiPair) == address(0),
+            address(bentoBox) == address(0),
             "strategy already initialized"
         );
-        require(address(IKashiPair(_kashiPair).bentoBox()) == _bentoBox);
-        require(address(IKashiPair(_kashiPair).asset()) == address(want));
+        require(_kashiPairs.length <= maxPairs, "exceeded maxPairs");
+        require(
+            _kashiPairs.length == _pids.length,
+            "kashiPairs and pids must be same length"
+        );
 
         bentoBox = IBentoBox(_bentoBox);
-        kashiPair = IKashiPair(_kashiPair);
+
+        for (uint256 i = 0; i < _kashiPairs.length; i++) {
+            kashiPairs.push(
+                KashiPairInfo(IKashiPair(_kashiPairs[i]), _pids[i])
+            );
+            require(
+                address(kashiPairs[i].kashiPair.bentoBox()) == _bentoBox,
+                "bento does not match"
+            );
+            require(
+                address(kashiPairs[i].kashiPair.asset()) == address(want),
+                "asset does not match want"
+            );
+
+            if (_pids[i] != 0) {
+                require(
+                    address(masterChef.poolInfo(_pids[i]).lpToken) ==
+                        _kashiPairs[i],
+                    "incorrect pid"
+                );
+
+                IERC20(_kashiPairs[i]).safeApprove(
+                    address(masterChef),
+                    type(uint256).max
+                );
+            }
+        }
 
         want.safeApprove(_bentoBox, type(uint256).max);
+        sushi.safeApprove(address(sushiRouter), type(uint256).max);
+
+        // Initialize the swap path
+        path = new address[](2);
+        path[0] = address(sushi);
+        path[1] = address(want);
     }
 
     function name() external view override returns (string memory) {
-        return
-            string(
-                abi.encodePacked(
-                    "StrategyKashiLender(",
-                    kashiPair.symbol(),
-                    ")"
-                )
-            );
+        return string(abi.encodePacked("StrategyKashiMultiPairLender"));
     }
 
     function estimatedTotalAssets() public view override returns (uint256) {
-        uint256 totalShares =
-            sharesInBento().add(
-                kashiFractionToBentoShares(kashiFraction(), true)
+        uint256 totalShares = sharesInBento();
+
+        for (uint256 i = 0; i < kashiPairs.length; i++) {
+            totalShares = totalShares.add(
+                kashiFractionToBentoShares(
```

Comment:
if we have less pairs than length, does this return 0 correctly?

Reply:
The `kashiPairs` are a dynamic array now, so it will. Is it better to use a fixed size array?

Reply:
my questions is the same, left a question below regarding the internal line 201 calc if it handles 0 totalShares correctly when strat is initialized

Reply:
@poolpitako @storming0x what cases would we have less pairs than length? If I understand the `push` and `pop` correctly, they should allows result in the array resizing.

Reply:
i think we are talking about a rare case where the strategy is iniatlized with 0 kashipairs, or it gets down to 0 manually so we ensure that this returns 0 at the end

Reply:
Added tests for both these cases, things seem to work fine.